correctme/ui/gui.py

```diff
@@ -1,146 +1,78 @@
-from os.path import join, dirname
-
 from kivy.app import App
-from kivy.uix.label import Label
 from kivy.uix.gridlayout import GridLayout
-from kivy.uix.textinput import TextInput
-from kivy.uix.button import Button
-from kivy.lang import Builder
-
-Builder.load_file(join(dirname(__file__), r'SimpleKivy.kv'))
+from kivy.properties import ObjectProperty
+from kivy.uix.widget import Widget
 
 from correctme import suggestForMe as sm
 
 
-class TextScreen(GridLayout):
-    prev1 = []
-    prev2 = []
-    prev3 = []
+class CorrectMeView(GridLayout):
+    input_text = ObjectProperty(None)
+    button1 = ObjectProperty(None)
+    button2 = ObjectProperty(None)
+    button3 = ObjectProperty(None)
+    button4 = ObjectProperty(None)
 
     def __init__(self, dataset, **kwargs):
-        super(TextScreen, self).__init__(**kwargs)
-        self.rows = 6
-        self.cols = 1
-        self.sentence = ""
-
-        self.health_label = Label(text="CorrectMe is here!!!")
-        self.add_widget(self.health_label)
-        self.username = TextInput(multiline=False)
-        self.username.bind(text=self.on_text)  # specify the callback function when the text changes for the textinput
-        self.username.keyboard_suggestions = True
-        self.add_widget(self.username)
+        super(CorrectMeView, self).__init__(**kwargs)
         self.tree, self.meta_dict = sm.initializeApp(dataset)
+        self.buttons = [self.button1, self.button2, self.button3, self.button4]
```

Comment:
Sorry if I am being nitpicky, but I think it's better if the buttons variable too is a class variable since it does not depend on the constructor's arguments. If there's any other reason for it being inside the constructor (which I am missing), then let's leave it as is.

Reply:
Thank you @piyush-jaiswal for this question. I did some research and it turns out this is the correct position for the `buttons` attribute. I wanted the list of the buttons which were defined in the `correctme.kv` file, and it turns out that to link the `correctme.kv` UI definitions we have to call the parent's `__init__` function, hence I need to make the list after that otherwise I would just have a list of 4 None Object properties. I am not sure whether this is the correct way or not. But for now this seems the only way.

Other way could have been not to define `button1`, `button2`, `button3` and `button4` and somehow connect the buttons to the list `buttons` directly in the `correctme.kv` file. But I am unable to figure out for now as I am very new to the `kv` syntax.

References:
1. [Kivy-connecting kv definitions](https://stackoverflow.com/questions/45699718/trouble-connecting-objects-in-kv-file-to-python-class)

2. [Class instances attributes](https://www.python-course.eu/python3_class_and_instance_attributes.php)

Reply:
I think when `CorrectMeView`'s parent's constructor is called, it automatically creates `button1`, `button2` etc. member variables inside `self`. Since we are using those variables to create the `buttons` list (and not the class variables) we can remove them completely.